apm-agent-plugins/apm-log-shipper-plugin/src/main/java/co/elastic/apm/agent/log/shipper/ApmServerLogShipper.java

```diff
@@ -51,12 +50,12 @@
     private File currentFile;
     private Set<TailableFile> tailableFiles = new HashSet<>();
 
-    public ApmServerLogShipper(ApmServerClient apmServerClient, ReporterConfiguration reporterConfiguration, MetaData metaData, PayloadSerializer payloadSerializer) {
-        super(reporterConfiguration, metaData, payloadSerializer, apmServerClient);
+    public ApmServerLogShipper(ApmServerClient apmServerClient, ReporterConfiguration reporterConfiguration, PayloadSerializer payloadSerializer) {
+        super(reporterConfiguration, payloadSerializer, apmServerClient);
     }
 
     @Override
-    public boolean onLineAvailable(TailableFile tailableFile, byte[] line, int offset, int length, boolean eol) throws IOException {
+    public boolean onLineAvailable(TailableFile tailableFile, byte[] line, int offset, int length, boolean eol) throws Exception {
```

Comment:
[question] why change to the broad `Exception` here ? If it's a way to make uses of this API to having to handle potential exceptions, that would be worth to add the the method Javadoc to be explicit.

Reply:
Because it felt inappropriate to add `UninitializedException` (serializer-specific) to the `FileChangeListener` interface...
Would you prefer that? Or something else?